apps/ewallet/lib/ewallet/exporters/csv_exporter.ex

```diff
@@ -61,11 +61,32 @@ defmodule EWallet.CSVExporter do
      }}
   end
 
+  def terminate(reason, %{export: %{status: "processing"} = export}) do
+    {:ok, _} =
+      AdapterHelper.store_error(export, "The server crashed during processing.", inspect(reason))
+  end
+
+  def terminate(reason, %{export: %{status: "new"} = export}) do
+    {:ok, _} =
+      AdapterHelper.store_error(export, "The server crashed during boot.", inspect(reason))
+  end
+
+  def terminate(_reason, _state), do: :ok
+
   def handle_cast(:upload, state) do
     adapter_module = get_adapter_module()
-    adapter_module.upload(state)
 
-    {:stop, :normal, state}
+    case adapter_module.upload(state) do
+      {:ok, export} ->
+        {:stop, :normal, %{state | export: export}}
+
+      {:error, %{status: "failed"} = export} ->
```

Comment:
Can you add a comment here to document why `{:error, %{status: "failed"}}` acts the same as `{:ok, export}` (i.e. does not call `AdapterHelper.store_error/2` like below)?